overlord/hookstate/ctlcmd/set.go

```diff
@@ -42,6 +56,51 @@ func (s *setCommand) Execute(args []string) error {
 		return fmt.Errorf("cannot set without a context")
 	}
 
-	// TODO: Talk to the handler to take care of the set request.
+	// Extract the transaction from the context. If none, make one.
+	s.context().Lock()
+	transaction, ok := s.context().Cached("transaction").(*configstate.Transaction)
```

Comment:
Should probably use what @pedronis described, with a local empty struct typed key.

Also note that this is specifically a configuration transaction. We don't want to have this idea observed globally as a "transaction".